### PR TITLE
[Community] Accepting/Rejecting membership requests when control node is offline

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Closing #11770 

Changes:
1. Adding Accepted/Rejected Pending in activity notification and community members settings
2. Adding storybook pages for `ActivityNotificationCommunityMembershipRequest` and `MembersTabPanel`
3. Update the `MembershipCta` design for Accept/Reject buttons 
4. Adding the new states in nim

Can be tested only in Storybook.

Not handled in this PR:
1. Link the backend implementation for the new states
2. Accepted/Rejected Pending states can be changed only by the admin that initiated the request. In this case the other options needs to be hidden in the UI

### Affected areas

Community members settings
Membership request notification

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/47811206/b71fba82-5bb6-4052-abbe-1e3b8681b10b

Screenshot 2023-08-07 at 16 27 42

### CODE REVIEW

File: ui/imports/utils/Constants.qml [hunk 1]
```diff
@@ -1113,7 +1113,19 @@ QtObject {
         MembershipRequests,
         RejectedMembers,
         BannedMembers
-    }   
+    }
+
+    enum CommunityMembershipRequestState {
+        Pending,
```
Comment (caybro) [inline ui/imports/utils/Constants.qml:1119]: This is imho missing (and should start with) 
```qml
None = 0,
```
so that it mirrors the NIM enum

**Comment by caybro** [comment]
The `MembersTabPanel` button alignment looks weird with the pending states, imho those should be right aligned; afaik you can't have both "Accept Pending" and "Reject Pending" at the same time right?

File: ui/app/AppLayouts/Communities/panels/MembersTabPanel.qml [hunk 2]
```diff
@@ -102,28 +103,46 @@ Item {
...
+                        id: rejectButton
+                        //using opacity instead of visible to avoid the acceptButton jumping around 
+                        opacity: ((root.panelType === MembersTabPanel.TabType.PendingRequests) && isHovered) ||
+                                    model.membershipRequestState === Constants.CommunityMembershipRequestState.RejectedPending
+                                    //TODO: Only the current user can reject a pending request, so we should check that here
+
+                        tooltipText: qsTr("Waiting for owner node to come online")
+                        interactive: model.membershipRequestState !== Constants.CommunityMembershipRequestState.RejectedPending
+                        buttonComponent: StatusButton {
+                            text: model.membershipRequestState == Constants.CommunityMembershipRequestState.RejectedPending ? qsTr("Reject pending") : qsTr("Reject")
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/MembersTabPanel.qml:138]:
```suggestion
                            text: model.membershipRequestState == Constants.CommunityMembershipRequestState.RejectedPending ? qsTr("Reject Pending") : qsTr("Reject")
```

same capitalization as above

File: ui/imports/utils/Constants.qml (hunk 1 above)
Comment (alexjba) [reply ui/imports/utils/Constants.qml:1119]: Good point, thanks!

**Comment by alexjba** [comment]
> same capitalization as above

> The `MembersTabPanel` button alignment looks weird with the pending states, imho those should be right aligned; afaik you can't have both "Accept Pending" and "Reject Pending" at the same time right?

@caybro I've forwarded these questions to Figma, but I don't think we're going to have a design update soon. WDYT of going with the current version and polish it when Ben is back?

**Review by caybro — APPROVED** [review]
LGTM

File: ui/app/mainui/activitycenter/panels/MembershipCta.qml [hunk 3]
```diff
@@ -14,6 +15,9 @@ Item {
     property bool pending: false
     property bool accepted: false
     property bool declined: false
+    property bool acceptedPending: false
```
Comment (noeliaSD) [inline ui/app/mainui/activitycenter/panels/MembershipCta.qml:18]: I know this is not new and you just followed the existing code but, I'm wondering if it makes sense to update all these state properties to be `readonly` and have only a new property called `membershipStatus` coming from outside. Something like this:
```
property int membershipStatus

readonly property int pending: membershipStatus === ActivityCenterStore.ActivityCenterMembershipStatus.Pending
readonly property int accepted: notification.membershipStatus === ActivityCenterStore.ActivityCenterMembershipStatus.Accepted
...

```

**Review by noeliaSD — APPROVED** [review]
LGTM! Tested on `storybook`.

Just added a suggestion and some doubts here:
- `Users are only moved into the relevant tab in the Members section (All members if join request accepted, Rejected if rejected) once the control node comes back online` --> Is this requirement covered or part of the backend integration?
- Is there already a task created for handling the integration missing?

File: ui/app/mainui/activitycenter/panels/MembershipCta.qml (hunk 3 above)
Comment (alexjba) [reply ui/app/mainui/activitycenter/panels/MembershipCta.qml:18]: good point!

**Pull request merged**